### PR TITLE
Added guide 'install jupyterlab the hard way' #2110

### PR DESCRIPTION
**Pull request by mangecoeur**
Following up on https://github.com/jupyterhub/jupyterhub/issues/2110, I've added the tutorial I created detailing installation of Jupyterhub in 'manual mode', based on a [blog post](https://medium.com/@mangecoeur/install-jupyterhub-and-jupyterlab-the-hard-way-14d6c134226a) I wrote.

It sure could be polished in various ways, I'm hoping by creating the PR it'll get some peer review. For context, this is the basis of the setup I run for our lab (~20 people sharing a high-end workstation machine)

### CODE REVIEW

**Comment by willingc** [comment]
@mangecoeur @GeorgianaElena Love the guide. I would change the title to something like: Installing JupyterHub and JupyterLab from the ground up or From Baremetal to JupyterHub+JupyterLab. The hard way tutorial was started to give a deep dive architectural understanding of JupyterHub similar to Kelsey Hightower's for K8s. There's definitely some of this that could be reused in the JupyterHub Hard Guide too.

**Comment by mangecoeur** [comment]
@willingc  @GeorgianaElena Thanks for the feedback. I changed the title (along with a few small corrections) as suggested.

I am happy to also contribute to the 'JH the hard way' repo, but @GeorgianaElena are you suggesting I do this *instead* of a guide in the main JH docs or *as well as*. Personally I prefer the second option since I think the main JH docs could benefit from a more in-depth guide, even if there is then a complete deep-dive guide in the separate repo.

File: docs/source/installation-guide-hard.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+This will produce the default configuration file `/opt/jupyterhub/etc/jupyterhub/jupyterhub_config.py`
+
+### Setup Systemd service
+
+We will setup Jupyterhub to run as a system service using Systemd (which is responsible for managing all services and 
+servers that run on startup in Ubuntu). We will create a service file in a suitable location in the virtualenv folder 
+and then link it to the system services. First create the folder for the service file:
+
+```sh
+mkdir -p /opt/jupyterhub/etc/systemd
```
Comment (GeorgianaElena) [inline docs/source/installation-guide-hard.md]: This command should be run with `sudo`.

File: docs/source/installation-guide-hard.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+Paste the following into the file:
+
+```
+[Unit]
+Description=Jupyterhub
+After=syslog.target network.target
+
+[Service]
+User=root
+Environment="PATH=/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/jupyterhub/venv/bin"
```
Comment (GeorgianaElena) [inline docs/source/installation-guide-hard.md]:
```suggestion
Environment="PATH=/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/jupyterhub/bin"
```

File: docs/source/installation-guide.rst [hunk 3]
```diff
@@ -11,3 +11,4 @@ running on your own infrastructure.
    quickstart
    quickstart-docker
    installation-basics
+   installation-guide-hard
```
Comment (GeorgianaElena) [inline docs/source/installation-guide.rst]: I believe the test is failing because there's a blank line missing at the end of this file.

**Comment by minrk** [comment]
Thanks everyone for the contributions and review! I do think there is a place in the main docs for a more complete installation step-by-step like this that is short of what we have in mind for "the hard way" in the dedicated tutorial repo, which should go much deeper. I think this PR hits the right balance for what belongs here in the main docs.

**Comment by mangecoeur** [comment]
Thanks for the revisions and comments, I've made some more changes to address comments and flesh out s few bits. Please let me know if there are further issues.

**Pull request merged**